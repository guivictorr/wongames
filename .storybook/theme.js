import { create } from '@storybook/theming'
import theme from '../src/styles/theme'

export default create({
  base: 'dark',

  colorPrimary: theme.colors.primary,
  colorSecondary: theme.colors.secondary,

  // UI
  appBg: theme.colors.mainBg,
  appContentBg: theme.colors.mainBg,
  appBorderColor: theme.colors.lightGray,
  appBorderRadius: 4,

  // Typography
  fontBase: '"Open Sans", sans-serif',
  fontCode: 'monospace',

  // Text colors
  textColor: theme.colors.white,
  textInverseColor: theme.colors.primary,

  // Toolbar default and active colors
  barTextColor: theme.colors.white,
  barSelectedColor: theme.colors.primary,
  barBg: theme.colors.mainBg,

  // Form colors
  inputBg: theme.colors.lightGray,
  inputBorder: theme.colors.darkGray,
  inputTextColor: theme.colors.black,

  brandTitle: 'Won games storybook',
  brandUrl: 'https://example.com',
  brandImage: '/img/logo.svg'
})
