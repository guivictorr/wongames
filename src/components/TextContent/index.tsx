import Heading from 'components/Heading'
import * as S from './styles'

export type TextContentProps = {
  title?: string
  content: string
}

const TextContent = ({ title, content }: TextContentProps) => (
  <S.Wrapper>
    {!!title && (
      <Heading line="left" lineColor="secondary">
        {title}
      </Heading>
    )}
    <div dangerouslySetInnerHTML={{ __html: content }}></div>
  </S.Wrapper>
)

export default TextContent
