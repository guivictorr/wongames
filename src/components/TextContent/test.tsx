import { screen } from '@testing-library/react'
import theme from 'styles/theme'
import { renderWithTheme } from 'utils/tests/helpers'

const props = {
  title: 'Description',
  content: `<h1>Content</h1>`
}
import TextContent from '.'

describe('<TextContent />', () => {
  it('should render the title and content', () => {
    const { container } = renderWithTheme(<TextContent {...props} />)

    expect(
      screen.getByRole('heading', { name: /description/i })
    ).toBeInTheDocument()
    expect(
      screen.getByRole('heading', { name: /content/i })
    ).toBeInTheDocument()

    expect(container.firstChild).toMatchSnapshot()
  })

  it('should render without title', () => {
    renderWithTheme(<TextContent content={props.content} />)

    expect(
      screen.queryByRole('heading', { name: /description/i })
    ).not.toBeInTheDocument()
  })

  it('should render change text correct color for mobile/desktop', () => {
    renderWithTheme(<TextContent {...props} />)
    const wrapper = screen.getByRole('heading', {
      name: /description/i
    }).parentElement
    expect(wrapper).toHaveStyle({
      color: theme.colors.white
    })

    expect(wrapper).toHaveStyleRule('color', theme.colors.black, {
      media: '(min-width: 768px)'
    })
  })
})
