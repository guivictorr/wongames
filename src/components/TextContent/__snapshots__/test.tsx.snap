// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<TextContent /> should render the title and content 1`] = `
.c1 {
  color: #FAFAFA;
  padding-left: 0.8rem;
  border-left: 0.7rem solid #3CD3C1;
  font-size: 1.8rem;
}

.c0 {
  color: #FAFAFA;
}

.c0 h1,
.c0 h2,
.c0 h3,
.c0 h4,
.c0 h5,
.c0 h6 {
  color: #FAFAFA;
  margin: 1.6rem 0;
  padding-left: 0.8rem;
  border-left: 0.7rem solid #3CD3C1;
}

.c0 p {
  margin-bottom: 1.6rem;
}

.c0 a {
  color: #F231A5;
}

.c0 img {
  max-width: min(70rem,100%);
  margin-bottom: 1.6rem;
}

.c0 ul,
.c0 ol {
  padding: 1.6rem 2.4rem;
}

.c0 hr {
  margin: 2.4rem 0;
}

@media (min-width:768px) {
  .c1 {
    font-size: 2.8rem;
  }
}

@media (min-width:768px) {
  .c0 {
    background: #FAFAFA;
    padding: 3.2rem;
    color: #030517;
  }
}

@media (min-width:768px) {
  .c0 h1,
  .c0 h2,
  .c0 h3,
  .c0 h4,
  .c0 h5,
  .c0 h6 {
    color: #030517;
  }
}

<div
  class="c0"
>
  <h2
    class="c1"
  >
    Description
  </h2>
  <div>
    <h1>
      Content
    </h1>
  </div>
</div>
`
